### PR TITLE
Fix dynamic workflows serialization

### PR DESCRIPTION
**Pull request by narape**
# TL;DR
Fix dynamic workflow task serialization. We broke dynamic workflows when we added strong types to workflows. 

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Dynamic workflows output a [DynamicJobSpec](https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/core/dynamic_job.proto#L12) that gets written in the propeller bucket as `futures.pb`, then propeller read this file and executes the workflow.

The problem was that we changes how we register outputs with the strong types and this spec proto didn't contain the binding for the outputs, then the written file was missing this information and propeller failed and retried the task until the retry counts reach its limit,

We didn't notice the breakage because we didn't contain unit tests for that (fixed) and the integration test was disabled (now enabled as well)

## Tracking Issue
NA

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit-java/src/main/java/org/flyte/flytekit/SdkDynamicWorkflowTaskRegistrar.java [hunk 1]
```diff
@@ -82,12 +83,21 @@ public DynamicJobSpec run(Map<String, Literal> inputs) {
       SdkWorkflowBuilder builder = new SdkWorkflowBuilder();
 
       InputT value = sdkDynamicWorkflow.getInputType().fromLiteralMap(inputs);
-      sdkDynamicWorkflow.run(builder, value);
+      var output = sdkDynamicWorkflow.run(builder, value);
```
Comment (honnix) [inline flytekit-java/src/main/java/org/flyte/flytekit/SdkDynamicWorkflowTaskRegistrar.java:86]: Have we agreed to start using `var` for everything? If so, maybe should do a big refactoring PR later.

**Review by andresgomezfrr — APPROVED** [review]

**Review by pablocasares — APPROVED** [review]

File: flytekit-java/src/main/java/org/flyte/flytekit/SdkDynamicWorkflowTaskRegistrar.java (hunk 1 above)
Comment (bergman) [reply flytekit-java/src/main/java/org/flyte/flytekit/SdkDynamicWorkflowTaskRegistrar.java:86]: I don't think so, let's revert/explicit type
